### PR TITLE
IAA Cargo Access Sanity Tweak

### PR DESCRIPTION
**Pull request by Optimism333**
<tweak>

## The Point of this PR ##
- The IAA job has had unusable cargo bay access for quite some time now. Presumably, it's a leftover from a time when the job had maintenance access. So IAAs could do surprise inspections by creeping into the cargo bay from maint like a weirdo. At least, that's what I assume the point of it was. However, when the job's maintenance access was removed, the inner cargo bay access was left untouched. So it's just been there, a pointless piece of access that can't be utilized anymore. I decided to make this pr as a possible solution.

## What Does the PR Do? ##
- This pr removes the old inner cargo bay access and replaces it with access to the cargo front desk instead. I think this is the solution that makes the most sense, since the IAA meeting with techs or the QM at the front and/or checking the paperwork in the filing cabinet is something that I could see them having the access for. However, I'm willing to consider other alternatives depending on feedback to the pr.
- Tested locally and it works as intended.

## Other Alternatives ##
- Just removing the cargo access outright, no other changes.
- Adding cargo office access and leaving the cargo bay access untouched. 
- Changing the ID numbers on the cargo front doors through mapping. (Suggested by BurntDevil)

:cl:
 * rscdel: IAA has lost their access to the inner cargo bay.
 * rscadd: IAA now has access to the cargo front desk.

### CODE REVIEW

**Comment by Kromkar** [comment]
Why not just remove it? Odd for a lawyer alt title to be able to enter another department. Why bot give them front access to medical, engineering, etc. As well?

**Comment by Optimism333** [comment]
> Why not just remove it? Odd for a lawyer alt title to be able to enter another department. Why bot give them front access to medical, engineering, etc. As well?

I'm only taking cargo access that already existed for the job and repurposing it to actually be usable, not making an argument to give them access to other departments where no access was present prior to this pr. Access is best handled in a minimalistic way, after all.

As for a justification to why they might have access to the front desk, I'd go by the idea that they might audit the cargo paperwork (which can be stored at the front desk filing cabinet) to make sure orders aren't being stolen. This does occasionally come up when people accuse cargo of stealing orders and the HOP and sec have bigger fish to fry.

Plus, they already do have access to the front door of medical. Can't give them what they already have!

**Comment by Optimism333** [comment]
I do agree that the Lawyer alt-title has weak justification for having all the IAA's accesses (I don't think they should have anything but law office, courtroom, and security access), but I see that as more of a problem with the alt-title itself.

**Comment by Eneocho** [comment]
The changelog's first point should be rscdel, not rscadd. You removed the inner cargo access, after all.

**Comment by BurntDevil** [comment]
IAA Cargo inspections used to be my bread and butter gimmick when I still played IAA, before I made Bridge Officer, and it'd be nice to be able to go into cargo still. Just make sure they can still use the cargo computer, or youre gonna mess that part of BO up.

**Comment by Optimism333** [comment]
> IAA Cargo inspections used to be my bread and butter gimmick when I still played IAA, before I made Bridge Officer, and it'd be nice to be able to go into cargo still. Just make sure they can still use the cargo computer, or youre gonna mess that part of BO up.

I just messed around a bit locally to try to understand what you meant. Do you mean that with the previous inner cargo bay access an IAA/BO could order and approve cargo orders using the supply console on the bridge? Because if that's the case, this pr does indeed change that.

An IAA/BO can still put orders into the supply shuttle console after this change, but someone in cargo or the HOP will need to approve them. I don't see anything wrong with this, though. It makes sense to me that the movement of cargo should be regulated by actual members of the department or one of its supervisors, not by an IAA/BO.

**Comment by BurntDevil** [comment]
That complicates alot of thing I normally do as BO, especially on deadpop. Why dont you, instead of changing the ID access of IAA, change the ID code on the doors, so it doesnt change anything else?

**Comment by Optimism333** [comment]
> That complicates alot of thing I normally do as BO, especially on deadpop. Why dont you, instead of changing the ID access of IAA, change the ID code on the doors, so it doesnt change anything else?

It seems weird to balance changes like this with deadpop in mind because in my experience handing out extra access with the spare is often the norm on deadpop anyways. Assuming that the station has a reasonably staffed crew, why would the BO need the access to move and approve cargo?

I'll try and find the ID code and see how it works.

**Comment by BurntDevil** [comment]
Its as simple as opening the map editor and matching the ID number on the lobby door with the back room's door. This has the least amount of impact on other aspects of the roles while accomplishing your task.

**Comment by Optimism333** [comment]
> Its as simple as opening the map editor and matching the ID number on the lobby door with the back room's door. This has the least amount of impact on other aspects of the roles while accomplishing your task.

Thanks for the tip. However, I still would like to know your answer to my question. 

Why does a BO need the ability to approve cargo orders, assuming the station has a normal number of staff? Isn't that the domain of the HOP and the actual cargo department?

I just don't see why that would be something that's necessary in their round start access.

**Edit:** I also just thought of something else. Wouldn't changing the ID access on the cargo lobby door make the "cargo office" access on the HOP computer misleading? Because then it wouldn't actually give access to the cargo office, the "cargo bay" access would do that instead.

**Comment by BurntDevil** [comment]
Its a redundant access type. Everyone else who has cargo access has access to all of cargo. We can get rid of one of the two without changing any existing gameplay. I would really prefer you do that than mess with BO gameplay. I use that alot to help heads of staff out, especially when theres no cargo personnel or I need to react quickly to a HoS request or the such. Part of BO is doing yeoman duties, the supply thing is part of that and has been a core part of my routine since I made BO.

**Comment by Optimism333** [comment]
> Its a redundant access type. Everyone else who has cargo access has access to all of cargo. We can get rid of one of the two without changing any existing gameplay. I would really prefer you do that than mess with BO gameplay. I use that alot to help heads of staff out, especially when theres no cargo personnel or I need to react quickly to a HoS request or the such. Part of BO is doing yeoman duties, the supply thing is part of that and has been a core part of my routine since I made BO.

I can understand you're not happy that this messes with something you see as part of a BO's gameplay, and I'm not unsympathetic. 

However, I still don't think the access to approve cargo orders is something that's necessary to the job or any of the alt-titles. I'm not convinced that such access shouldn't be restricted to the department directly responsible for cargo, and balancing round-start access under the assumption that there might not be members of the department doesn't seem reasonable. Neither does the idea that another role that's not part of the department can do their job faster and should have the access to do it. 

As for the example you provided, the only thing that this would slow down is that after the BO puts the cargo orders into the bridge cargo console, someone in cargo or the HOP will have to approve them. That's a single additional communication that has to be sent via PDA or radio, assuming there's an active cargo department. Hardly a huge delay, and the BO can still play a part in expediting it by putting the orders in.

There's ways to get around this new restriction on deadpop as well, since you previously brought that up. The spare is the first and easiest way to get the extra cargo access. And if the spare isn't available for any reason, a soldering iron can be used on the circuit board for the cargo console to override the access restrictions. The second option only requires a few common tools and some sulphuric acid.

tldr: Respectfully disagree, but I appreciate that you took the time to write feedback. I encourage you to downvote the pr if you really don't like this change.

**Comment by BurntDevil** [comment]
I just dont think we should be changing something with one intent and doing it in a way that cascades changes down the line in other areas. Its outside of the scope of the original intent of the PR, which while I guess not "unatomic", certainly feels that way.

If the end goal is to simply allow IAA access into cargo, all attempts to do that should be taken to impact other areas of the game as little as possible if not at all. So again, tossing out the redundant layer of access between the two halves of cargo and configuring the doors accordingly does this without going beyond the scope of the PR's intent.

**Comment by Kurfursten** [comment]
I'm amazed how few people gave opinions given I specifically advertised this one in multiple places. I guess only the few actual IAA players will notice. It does look like BD is the only one against this though. If it does become intractable you can always push for your own solution, trying to rally votes behind a PR with map changes and reverting access changes.
